RDSC-1456 Add default job information (#2784)
RedisLabs/redislabs-docs: content/rdi/data-transformation/data-transformation-pipeline.md

![Data transformation pipeline, high level](/images/rdi/data-transformation-pipeline.png)

> ## Default Job

Reviewer: Default Job --> Default job (docs conventions)

> In situations where there is a need to perform a default data transformation, the default job is used. The transformation associated with this job will be applied to all tables that lack their own explicitly defined jobs. The default job must have a table name of "*", and only one instance of this type of job is permitted.

Reviewer 2: ```suggestion
In situations where there is a need to perform a transformation on all ingested record without creating a specific job for specific tables, the default job is used. The transformation associated with this job will be applied to all tables that lack their own explicitly defined jobs. The default job must have a table name of "*", and only one instance of this type of job is permitted.
```

Author: Modified to suggested.

Author: Modified

> For example, the default job can be used to implement common things like adding a prefix/postfix to all Redis keys or injecting a new field for each new hash or JSON written to the target without the need to define such transformations individually for each source table.

Reviewer 3: ```suggestion
For example, the default job can be used to implement common things like adding a prefix/postfix to all Redis keys, or injecting a new field for each new hash or JSON written to the target without the need to define such transformations individually for each source table.
```

Author: Added comma

> At this moment the default job is supported for 'ingest' pipelines only.

Reviewer 3: ```suggestion
At this moment the default job is supported for ingest pipelines only.
```

Reviewer 2: At this moment -> Currently

Author: Changed

> Currently the default job is supported for ingest pipelines only.

Reviewer 3: ```suggestion
Currently, the default job is supported for ingest pipelines only.
```

### Example

> This example demonstrates the process of adding an app_code field with a value of 'rdi' using the `add_field`` block to all tables that lack explicitly defined jobs. Additionally, it appends an 'aws' prefix and a 'gcp' postfix to every generated hash key.

Reviewer 3: The fragment "... using the `add-field` block to..." just doesn't read well. As I'm not sure what the intent is here, can you re-word this? I did fix up some quoting stuff below.

```suggestion
This example demonstrates the process of adding an `app_code` field with a value of `rdi` using the `add_field` block to all tables that lack explicitly defined jobs. Additionally, it appends an 'aws' prefix and a 'gcp' postfix to every generated hash key.
```

Reviewer 2: please use another value than 'rdi' in the example as it will just create confusion
use 'foo'

Author: Removed extra characters

> This example demonstrates the process of adding an `app_code` field with a value of `foo` using the `add_field` block to all tables that lack explicitly defined jobs. Additionally, it appends an `aws` prefix and a `gcp` postfix to every generated hash key.

Reviewer: Add link to `add_field` block reference in the docs.

Author: Done
